components/Modal.jsx

```diff
@@ -0,0 +1,34 @@
+import { useRef, useEffect } from "react";
+import styles from "@/styles/components/modal.module.scss";
+
+export default function Modal({ children, isOpen, setIsOpen }) {
+	const ref = useRef(null);
+
+	useEffect(() => {
+		if (isOpen) {
+			ref.current.showModal();
+		}
+	}, [isOpen]);
+
+	const handleBackgroundClicked = (event) => {
+		if (event.target === ref.current) {
+			setIsOpen(false);
+		}
+	};
+
+	const handleButtonClicked = (event) => {
+		if (event.key === "Escape") {
+			setIsOpen(false);
+		}
+	};
+
+	if (!isOpen) {
+		return;
```

Comment:
I haven't noticed it and I know we talked about this subject 😅
but... consider this >w<
```js
if (!isOpen) return;
```

Reply:
noted